### PR TITLE
@types/jsreport-core - template registry

### PR DESCRIPTION
**Pull request by newbish**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## jsreport-core ([<kbd>unpkg</kbd>](https://unpkg.com/browse/jsreport-core@latest/))
was missing the following properties:
1. Reporter
## jsreport ([<kbd>unpkg</kbd>](https://unpkg.com/browse/jsreport@latest/))
was missing the following properties:
1. Reporter
1. renderDefaults
1. render
1. extendDefaults
1. reporter

<details>
<summary>as well as these 49 other properties...</summary>
<p>core, Reporter, renderDefaults, render, extendDefaults, reporter, core, Reporter, renderDefaults, render, extendDefaults, reporter, core, Reporter, renderDefaults, render, extendDefaults, reporter, core, Reporter, renderDefaults, render, extendDefaults, reporter, core, Reporter, renderDefaults, render, extendDefaults, reporter, core, Reporter, renderDefaults, render, extendDefaults, reporter, core, Reporter, renderDefaults, render, extendDefaults, reporter, core, Reporter, renderDefaults, render, extendDefaults, reporter, core</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 65eb157d45b628d088a334dc4d9f47609b5770c1
</p>

File: types/jsreport-chrome-pdf/jsreport-chrome-pdf-tests.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ const jsreport = JsReport({
 });
 
 jsreport.beforeRenderListeners.add('test', (req, res) => {
-    console.log('input', req.template.content);
+    console.log('input', (req.template as JsReport.Template).content);
```
Comment (taoqf) [inline types/jsreport-chrome-pdf/jsreport-chrome-pdf-tests.ts:13]: I don't think it's a good idea to use `as` here.
Is there a better way?
I really do not have much time on this.

Comment (newbish) [reply types/jsreport-chrome-pdf/jsreport-chrome-pdf-tests.ts]: reading through some of the other extensions it appears that even when it's not explicitly defined it is still required that the content field have data.

https://github.com/jsreport/jsreport-chrome-pdf/blob/7b5ea5b14d14219f37ad254f4e879550732701ed/lib/chrome.js#L18
https://github.com/jsreport/jsreport-docx/blob/b50fad34b38f55b01eaa9093aa1192749b6ca21f/lib/docx.js#L27

Comment (pofider) [reply types/jsreport-chrome-pdf/jsreport-chrome-pdf-tests.ts]: The `template.content` is required when you use just `jsreport-core` with lets say `jsreport-handlebars` extension.  However in typical cases, when you use more extensions, the `template.content` gets optional. 

Two examples
`jsreport-template` - lets the caller specify `template.name | shortid` and replace the input `template` with entity from db, so the `template.content` isn't required
`jsreport-docx` - loads the `template.content` from the linked word asset, so the `template.content` isn't required

And there are other examples. 

Thank you for your work here. Don't hesitate to mention me here if you have questions.

**Comment by newbish** [comment]
There appears to be an issue with Typescript 4.2 that's unrelated to the lib

**Pull request closed without merging**